staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -282,6 +289,49 @@ public void cleanUpStaleInprogressBatches()
         }
     }
 
+    @Override
+    public BatchStatusResponse getBatchStatus(final TenantId tenantId, final BatchId batchId) throws BatchNotFoundException
+    {
+        boolean batchInCompletedState = false;
+        BatchStatusResponse responseBean = new BatchStatusResponse();
+        responseBean.setBatchID(batchId.getValue());
+        BatchStatus batchStatus = new BatchStatus();
+        InProgress inProgress = new InProgress();
+        if (batchPathProvider.getPathForBatch(tenantId, batchId).toFile().exists()) {
+            batchStatus.setBatchComplete(true);
+            batchInCompletedState = true;
+        } else {
+            batchStatus.batchComplete(false);
+        }
+        List<InProgressMetrics> inProgressMetricsList =
+                new ArrayList<>();
+        //get threadIDs for requested batchId from filesystem
+        List<String> listOfInProgressBatches = batchPathProvider.getListOfInProgressThreadFromFileSystem(tenantId, batchId);
+        if(!batchInCompletedState && listOfInProgressBatches.isEmpty()) {
+            throw new BatchNotFoundException(batchId.getValue());
+        }
+        Map<String, Tracker> trackerMap = BatchProgressTracker.getInProgressTrackerMap();
+        listOfInProgressBatches.forEach(batchThread -> {
+            InProgressMetrics inProgressMetrics= new InProgressMetrics();
+            if (!batchThread.contains(ServiceIdentifier.getServiceId()))
+                batchPathProvider.monitorBatchProgressInFileSystem(tenantId, batchId, batchThread);
+            if(trackerMap.get(batchThread) != null) {
+                inProgressMetrics.setLastModifiedDate(trackerMap.get(batchThread).getLastModifiedTime());
+                inProgressMetrics.setBytesReceived(trackerMap.get(batchThread).getNumberOfBytesReceived());
+                inProgressMetrics.setBytesPerSecond(trackerMap.get(batchThread).getFileUploadRate());
+                inProgressMetricsList.add(inProgressMetrics);
+            }
+        });
+        // Remove the entry from TrackerMap for batch processed by different staging service
+        trackerMap.entrySet().removeIf(entry ->!entry.getKey().contains(ServiceIdentifier.getServiceId()));
+        // Build and return the Response
+        inProgress.setNumberOfBatches(inProgressMetricsList.size());
```

Comment:
I think we should drop this number of batches from the response payload.  The caller can easily see the size of the array themselves so it's really not adding anything.

Reply:
Done.